RomanNumeralsCI/src/main/java/com/ericsson/peterbuki/RomanConverter.java

```diff
@@ -4,43 +4,83 @@
  * Created by epetbuk on 2016.05.18..
  */
 public class RomanConverter {
-
+    public static final int MAX_VALUE = 3999 ;
 
     protected RomanConverter()
     {
         System.out.println("RomanConverter constructor.") ;
-        for (RomanNumber n : RomanNumber.values())
+        for (RomanNumberEnum n : RomanNumberEnum.values())
         {
             System.out.println(n.name + " = " + n.value) ;
         }
     }
 
     public static int convertToArabic(String in)
     {
+        boolean wasLastDouble ;
         int result = 0 ;
-        String toDecode ;
-        for (int position = 0; position < in.length(); position++)
+        for (int position = 0; position < in.length(); position+=wasLastDouble?2:1)
         {
-            RomanNumber rn = RomanNumber.Z ;
+            wasLastDouble=false ;
+            RomanNumberEnum rn = RomanNumberEnum.Z ;
 
             // try to convertToRoman 2 characters
             if (in.length()-position > 1)
-                rn = RomanNumber.findByName(in.substring(position, position+2)) ;
+                rn = RomanNumberEnum.findByName(in.substring(position, position+2)) ;
 
             // converted 2 characters, pos increased by additional 1
-            if (rn != RomanNumber.Z)
-                position++ ;
+            if (rn != RomanNumberEnum.Z)
+                wasLastDouble=true ;
             else
-                rn = RomanNumber.findByName(in.substring(position, position+1)) ;
+                rn = RomanNumberEnum.findByName(in.substring(position, position+1)) ;
 
             result += rn.value ;
         }
         return result ;
     }
+    public static boolean isArabic(String in)
```

Comment:
This method does something different that its name suggests. It checks if the string can be parsed and is below a certain value. It is better to have a separate method for the parsing. Use the option structure from java8 or this: https://github.com/google/guava/wiki/UsingAndAvoidingNullExplained